### PR TITLE
Adds gym scenario. Adds relevant professions and items. Adds new map location.

### PR DESCRIPTION
**Pull request by ghost**
#### Summary

SUMMARY: Content "Adds gym scenario, relevant professions, items and uncrafts"

#### Purpose of change

Exercise is a passion of mine. I wanted to bring my world into CDDA

#### Describe the solution

Added new scenario: The Gym with 3 starting locations.
Added 3 unique professions playable on The Gym scenario.
Added supplementary existing professions to selection menu of The Gym.
Added new map location. Populated.
Modified 2 existing gym layouts to accommodate new items
Modified slingshot recipe to allow use of a resistance band as a component

Added several new items:
1lb dumbbell, 5lb dumbbell, 10lb dumbbell, 20lb dumbbell
2.5lb plate, 5lb plate, 10lb plate, 25lb plate, 45lb plate
Resistance band, exercise ball
Created uncraft recipes for respective items

Added new furniture:
plate rack, dumbbell rack

#### Describe alternatives you've considered

Going to the gym in real life.

#### Testing

Tested at each stage of content entry. Compared values with existing items. Attempted to deconstruct objects. Attacked enemies with objects. Isolated location lottery to ensure constant testing of individual start locations. Spawned items in. Climbed up and down map locations.

#### Additional context

More fun than playing.

Screenshot from 2020-07-04 03-14-04
Screenshot from 2020-07-04 03-14-23
Screenshot from 2020-07-04 03-16-17
Screenshot from 2020-07-04 03-17-17
Screenshot from 2020-07-04 03-17-43
Screenshot from 2020-07-04 03-17-56
Screenshot from 2020-07-04 03-18-08
Screenshot from 2020-07-04 03-18-18
Screenshot from 2020-07-04 03-21-04
Screenshot from 2020-07-04 03-21-51
Screenshot from 2020-07-04 06-16-19
Screenshot from 2020-07-04 06-19-58

### CODE REVIEW

File: data/json/professions.json [hunk 1]
```diff
@@ -4232,5 +4232,65 @@
...
     "items": { "both": [ "shorts", "heroin", "syringe" ], "male": [ "briefs" ], "female": [ "bra", "panties" ] },
     "addictions": [ { "intensity": 40, "type": "opiate" } ]
+  },
+  {
+    "type": "profession",
+    "ident": "yoga_instructor",
+    "name": "Yoga Instructor",
+    "description": "You were in the middle of downward dog. Now you need to get down and out.",
+    "points": 2,
+    "traits": [ "FAST_REFLEXES", "LIGHTEATER", "THINSKIN", "FLIMSY" ],
```
Comment (LaVeyanFiend) [inline data/json/professions.json:4242]: I fail to see how any of the traits are justified outside of massive stereotyping.

File: data/json/professions.json [hunk 2]
```diff
@@ -4232,5 +4232,65 @@
...
+    },
+    "flags": [ "SCEN_ONLY" ]
+  },
+  {
+    "type": "profession",
+    "ident": "runner",
+    "name": "Runner",
+    "description": "Forest Chump was your role model growing up. Time to make him proud.",
+    "points": 2,
+    "traits": [ "GOODCARDIO", "HUNGER", "FLEET", "FLIMSY" ],
```
Comment (LaVeyanFiend) [inline data/json/professions.json:4260]: I can see only GOODCARDIO as justified.

File: data/json/professions.json [hunk 3]
```diff
@@ -4232,5 +4232,65 @@
...
+    },
+    "flags": [ "SCEN_ONLY" ]
+  },
+  {
+    "type": "profession",
+    "ident": "weight_lifter",
+    "name": "Lifter",
+    "description": "You were miring your gainz in the mirror when you saw the commotion outside.",
+    "points": 4,
+    "traits": [ "WEAKSTOMACH", "SMELLY", "CLUMSY", "STRONGBACK", "ADRENALINE", "GOODCARDIO", "ANTIJUNK" ],
```
Comment (LaVeyanFiend) [inline data/json/professions.json:4284]: I can only see STRONGBACK and GOODCARDIO as justified.

File: data/json/scenarios.json [hunk 4]
```diff
@@ -517,19 +517,19 @@
     "points": 3,
     "description": "Despite all the soldiers, guns and minefields, the base you were on got overrun by the dead.  Everyone was ordered to fall back to the armory, but during all the chaos you got lost and you are now stuck in the warehouse all alone.  You are not sure if anyone made it to the armory, or if you are the last man alive.",
     "start_name": "Military Base Warehouse",
-    "allowed_locs": [ "sloc_military_base_warehouse" ],
-    "professions": [ "unemployed", "soldier", "specops", "bio_soldier", "bio_sniper", "labtech", "medic", "mili_pilot" ],
+    "allowed_locs": [ "mil_base_2g" ],
+    "professions": [ "unemployed", "soldier", "specops", "bio_soldier", "bio_sniper", "labtech", "medic" ],
```
Comment (LaVeyanFiend) [inline data/json/scenarios.json:521]: Outside of scope of PR.

File: data/json/scenarios.json [hunk 5]
```diff
@@ -517,19 +517,19 @@
...
     "flags": [ "CHALLENGE", "LONE_START" ]
   },
   {
     "type": "scenario",
-    "ident": "presort",
-    "name": "Crazy Party",
```
Comment (LaVeyanFiend) [inline data/json/scenarios.json:527]: Also outside of scope(also why, I like that start 😢).

File: data/json/professions.json [hunk 6]
```diff
@@ -4232,5 +4232,65 @@
...
+    "skills": [ { "level": 1, "name": "dodge" } ],
+    "items": {
+      "both": {
+        "items": [ "runner_bag", "b_shorts", "tank_top", "sneakers", "socks_ankle", "wristwatch" ],
+        "entries": [
+          { "group": "charged_cell_phone" },
+          { "group": "charged_mp3" },
+          { "item": "towel" },
+          { "item": "toothbrush_plain" },
+          { "item": "hairbrush" }
```
Comment (LaVeyanFiend) [inline data/json/professions.json:4270]: These should be moved out of "entries" and into "items".

File: data/json/professions.json (hunk 2 above)
Comment (LucasHoage) [reply data/json/professions.json:4260]: The body type of the average runner is slender. Excessive amounts of cardio will put a burden on the body's ability to conserve mass. To offset the increase in metabolism through extended long term activities, the body will begin consuming: stored sugars->stored fats -> stored muscle mass, in that order. Exercising frequently will increase your resting metabolic resting rate. Your caloric intake to maintain the levels of fitness will increase. The same result is true when you increase your body mass(Body building). The larger your body mass, the more calories required to maintain the same level of body mass.

I do agree with you on the flimsy flag however. I misread that and thought it translated to lower weight carried. I mixed it up with bad back. Although it would be true that the average runner is not capable of performing at the same strength levels of a non runner. They don't have enough muscle mass. What I should have done was create a new trait to reflect a %15 decrease in weight carried instead.

File: data/json/professions.json (hunk 1 above)
Comment (LucasHoage) [reply data/json/professions.json:4242]: Following up from another reply. The only one here I agree with you on is Flimsy. Maybe Thinskin. I needed a negative to offset the bonus to dodge.

The average yoga instructor is flexible, dexterous, slim. Regular yoga is not very taxing on the body's muscle system and does not equate to a significant adaptation neurologically through induced loading. Power yoga on the other hand is a fork of yoga that is specifically designed to be load inducing utilizing the body's weight and static holds to induce a neurological adaptation period.

In short, flexibility training consumes a lot less calories than loading the muscle groups does.

File: data/json/professions.json (hunk 3 above)
Comment (LucasHoage) [reply data/json/professions.json:4284]: I agree weak stomach was a poor choice. I was supposed to choose the weakened immune system option. Excessive amounts of loading of muscle groups will leave you in with a supressed immune system.

Smelly was probably a bad idea.
Clumsy was a bad idea.

Adrenaline is a good choice. Explosiveness in lifting requires you to have a lot of carbohydrates before doing a prolonged workout. Glucose is released during an adrenaline rush. Glucose is also consumed during explosive lifting. Power lifters and strongmen will eat high carbohydrate meals before and during workouts. Strongmen are more likely to consume junk food since their diet primarily relies on calorie count after meeting their macros.

The lifting profession is a simplified version of all of the types of lifters. There really isnt a need to specifically create a profession for a bodybuilder, lifter, power lifter, strong man, cross fit, calisthenics, etc. They all do share quite a few comonalities.

Outside of being a strongman, the average lifter doesn't consume junkfood. Meeting your macros on a calorie restricted diet (cut) or even on a +500 calorie intake (bulk) would be jeopardized by eating junkfood. Outside of a few niche aspects of lifting (strongman, stone lifting, log throwing), junkfood is not a staple of the average lifter's diet.

File: data/json/scenarios.json (hunk 4 above)
Comment (LucasHoage) [reply data/json/scenarios.json:521]: I did not make that change. I'm not sure what is causing that honestly. I've had a few of these pop up and I think the best solution is to create a backup of my changes and then sanitize the results after repopulating the master.

Comment (LucasHoage) [reply data/json/scenarios.json:521]: Unless that has to do with me not appending my changes at the end of the file

File: data/json/scenarios.json (hunk 5 above)
Comment (LucasHoage) [reply data/json/scenarios.json:527]: I did not remove that

Comment (LucasHoage) [reply data/json/scenarios.json:527]: I don't understand why that happened.

File: data/json/professions.json (hunk 6 above)
Comment (LucasHoage) [reply data/json/professions.json:4270]: 10-4

File: data/json/professions.json (hunk 1 above)
Comment (LucasHoage) [reply data/json/professions.json:4242]: I will also try to supply screenshots. I am pooped right now. I will follow up.

Comment (LucasHoage) [reply data/json/professions.json:4242]: Pictures updated.

File: data/json/furniture_and_terrain/furniture-storage.json [hunk 7]
```diff
@@ -930,5 +930,53 @@
...
+      "sound_fail_vol": 5,
+      "items": [ { "item": "scrap", "count": [ 2, 8 ] }, { "item": "steel_chunk", "count": [ 2, 4 ] }, { "item": "pipe", "count": 1 } ]
+    }
+  },
+  {
+    "type": "furniture",
+    "id": "f_plate_rack",
+    "name": "plate rack",
+    "looks_like": "f_rack",
+    "description": "A metal rack designed to hold plated shaped weights",
```
Comment (matskuman5) [inline data/json/furniture_and_terrain/furniture-storage.json:963]:
```suggestion
    "description": "A metal rack designed to hold plated shaped weights.",
```

File: data/json/items/fluff.json [hunk 8]
```diff
@@ -235,5 +235,167 @@
...
+    "longest_side": "99 mm",
+    "bashing": 4,
+    "ammo_type": "thrown",
+    "thrown_damage": [ { "damage_type": "bash", "amount": 11 } ]
+  },
+  {
+    "type": "GENERIC",
+    "id": "plate_10lb",
+    "name": { "str": "10lb plate" },
+    "description": "A decently weighted disc shaped weight used on a weight lifting bar. Can be used as a bashing weapon. It is to awkward to throw.",
```
Comment (matskuman5) [inline data/json/items/fluff.json:353]:
```suggestion
    "description": "A decently weighted disc shaped weight used on a weight lifting bar. Can be used as a bashing weapon. It is too awkward to throw.",
```

Comment (AMurkin) [reply data/json/items/fluff.json:353]: Use double sentence spacing after periods. [doc/MANUAL_OF_STYLE.md](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/MANUAL_OF_STYLE.md)

File: data/json/furniture_and_terrain/furniture-storage.json (hunk 7 above)
Comment (LucasHoage) [reply data/json/furniture_and_terrain/furniture-storage.json:963]: 10-4 buddy. I just got home and I'm on furlough all next week. I'm gonna get on this. I have a few more things to change in that as well. Thanks for pointing it out.

File: data/json/items/fluff.json (hunk 8 above)
Comment (LucasHoage) [reply data/json/items/fluff.json:353]: 10-4 buddy. I'm on it this weekend. Thanks for your suggestions.

**Comment by LucasHoage** [comment]
I've resolved what I think are most of the issues regarding this PR. I do want feedback on the professions that I created. I also want feedback on the other portions of this project. I cannot add intended negative traits for balancing purposes yet as they haven't been committed yet. They are in: #41837

Muscle Atrophy and Suppressed Immunity.

**Comment by Night-Pryanik** [comment]
I'm not directly opposed to the idea of this new scenario, but at first glance it looks like just another location for the pre-existing Big Building scenario. What distinctive features your scenario has to justify its existence as a separate scenario?

**Comment by LucasHoage** [comment]
> I'm not directly opposed to the idea of this new scenario, but at first glance it looks like just another location for the pre-existing Big Building scenario. What distinctive features your scenario has to justify its existence as a separate scenario?

That's a good question and it has me now questioning whether or not the scenario even belongs in the game now. I want your input after this.

`[ "sloc_mall_loading_area", "sloc_mall_food_court", "sloc_apartments_rooftop", "sloc_hospital" ],`

`"flags": [ "SUR_START", "CITY_START", "LONE_START" ]`

The uniqueness of the Big Building scenario relies exclusively on physically large buildings. The size of the buildings in my scenario are not that large. My scenario also is not invoking the surrounded start flag which increases the difficulty significantly in an already difficult environment.

The goal of this scenario is not to introduce artificial difficulty levels. It is it reuse the existing default situations and combine them into a less combative experience and a more reasonable survival experience. Based on the design documentation, the start of the canon is suggested at day 45. In an attempt to get close enough to that time frame, I utilized the spring start flag:

`"flags": [ "LONE_START", "CITY_START", "SPR_START" ]`

No other scenario utilizes that flag. It forces the time to day 1 of spring. It's a happy compromise I was willing to make to get 2 goals reached:

Stay as close to the design document as possible
Introduce environmental hazards

The first goal was reached. The second goal required that I use roleplaying in tandem with the environmental hazard. In real life situations at the gym (Based on experience, not a data driven fact), most people who attend the gym are not comfortable with bringing a change of clothes and undressing in the locker room or even taking a shower at these establishments.

Knowing this, I had an answer for the environmental hazard. Give the new starting professions gym clothes. Because the temperature is substantially cold, not winter, but still a danger, the clothes the gym goer is wearing are not suitable for the environment. This causes a desperate need for the player to do 2 things immediately:

Avoid the monsters outside
Find warm clothes

So that brings us to the next issue at hand. How does the character deal with the monsters? Well, depending on your class choice, you can choose to deal with them utilizing skills, or you can use the new items that were spawned in the gyms (Depending on location). The resistance band can be turned into a slingshot provided you can get access to a nearby house quickly. This is a game changer early game and immediately improves the survive chances.

Alternatively, if the player chooses a strong, melee oriented character, they have an advantage using the dumbbells or the plate weights as makeshift weapons. Especially if they get lucky and roll a 45lb plate due to it's stunning and blocking abilities. But it IS heavy so it's not a long term feasible weapon. It's a means to survive until you can compound on the escape vector.

The environmental danger provides a compounding issue with weight as well. The cold makes it harder to move. So depending on your profession choice, it might be a good idea to instead take a smaller weapon from inside the gym to balance out the cold effects.

The gym provides an opportunity for people to roleplay in what I hope, is the closest representation of what the average gym professions I added are.(To be fair, some of the negative traits that do exist predominantly in real life situations do not exist in game yet.) Not only that, they are able to play in a building model based off of a real world gym in my hometown. With exception to a few items, the framework is identical. I think it being in the game will give a more in depth role play aspect, although short and temporary. Most of the game is not played in one place regardless.

Lastly, the new items can only be found in the commercial gyms. They won't be spawned anywhere else unless someone else decides to make that happen. I didn't want to take away incentive to use gyms with the addition of functional exercise equipment. They would be the easiest places to find a fully functional gym and reward the player for clearing those areas out either in this scenario or in alternatives.

Given that we're not forcing anything out of the ordinary or artificially inflating the difficulty barrier, I think that sets it apart quite nice. I want to know what you think. Be blunt if you have to.

**Comment by Night-Pryanik** [comment]
If I understood you correctly, you want your scenario to make accent on environmental hazards. And thus this scenario shouldn't be nailed down to the gym as the only and specific starting location. If so, you might want to generalize the scenario and rename it respectively and allow for other starting locations. For those who want some challenges, but don't want the harsh Really Bad Day scenario.

**Comment by LucasHoage** [comment]
> If I understood you correctly, you want your scenario to make accent on environmental hazards. And thus this scenario shouldn't be nailed down to the gym as the only and specific starting location. If so, you might want to generalize the scenario and rename it respectively and allow for other starting locations. For those who want some challenges, but don't want the harsh Really Bad Day scenario.

No, the primary goal was to remain as close as cannon as possible. The environmental hazard was a byproduct of existing functions of SPR_START that this scenario ultimately had a benefit from. In the end, if we removed the environmental hazard, what would be left is what is the most important aspect of any content and that is variety, which I'll go into further in a few. 

Before I do that, I wanted to also emphasize that the roleplay is a component as-well. I'm just seriously limited in with the current tools to do anything with that other than the flyer I made in a different PR that advertises for the gym. There are a few scenarios that utilize flavortext to assign the time of the scenario, such as FoodPerson which describes the time as happening right at the start of the Cataclysm. It could be argued very easily that electricity should still be working during that moment in time. Currently this does not work as it has not been built in the game.( Not how it is intended to work in conventional situations.) I would have liked to created a PA Speaker box item that announces in intervals things like scheduled training sessions with your local trainer or membership discounts when you opt in for a newsletter with promotional offers, etc.

Outside of creating a more realistic gym based on current interior decorating standards backed by marketing analysis, there's not a whole lot to do with the tools that I have and with the gym scenario at all. Physical fitness is an action oriented, visually driven experience. When people tuned in for the Arnold Strongman Santa Monica earlier this year, they were looking at 2 big contenders, Martins Licis and Brian Shaw. When you see people like this, they are physical anomalies compared to the average lifter, let alone the average person. They put on a show with their physical talents. Sometimes they fluff certain things like their breathing or grunting at certain times for dramatic effect for the crowd and the viewers at home.

Those types of things aren't supported the way the game currently is. It will be a long time before that could become a reality. The happy middle is to build the environment to facilitate those two most important things: Physical fitness equipment and the interior decoration to match what is data driven. One thing that really stands out to mind is mirrors. There are mirrors that are placed in very specific parts of a gym to maximize the attention lifters will give to a specific area. One of these places is the curl bar rack. When the lifter uses those bars, they ultimately want to see the fake temporary pump they're getting that makes their veins pop out. It looks nice, it drives sales indirectly. It's physically and visually driven.

These are nuances that again, cannot be placed into the game. Some things can but regardless, they wouldn't be unique to any given scenario, they would be unique to items with an action feedback after activating an object. I don't know where I'm going with this other than to see that Roleplay is an important aspect of gameplay and it is a driver for longevity.

This ties into what the rebuttal is about. I think your argument is more about categorization for the sake of uniqueness and maybe I'm interpreting this incorrectly, is suggesting nihilism with the question of "Do we need more variety when there already is a lot?"

The answer to the second question is going to be yes and forever will be yes.

To address what I think is the first concern you brought up about categorizing or structuring things to allow for all environmental hazards. I don't think this is the correct way to evaluate the situation.

If we break down certain aspects of each scenario, one thing they all have in common in submap coordinate modification. Each scenario has a list of starting locations. The shelter has 6 versions, all of them unique in their own respects. The randomness factor is a high driver of the uniqueness within each variant whereas we really have only two locations, vandalized and not. In the end it is just a building that has items that make the difficulty curve lower right? There is almost no challenge. It is a low barrier entry for new players. I would even argue that it's no different than the safe place start or even the sheltered survivor start. Both provide very easy means of survival and have absolutely minimal forced roleplay.

It has no modifiers that artificially increase the difficulty barriet such as SUR_START. It has a roleplay aspect based on flavortext on character generation. It has an unrestricted profession menu. It follows up with the visual and physical component by spawning a computer with more flavortext. It follows up with that by giving the player a goal to reach. If that computer was not there, it would just be another building with a submap coordinate modification right? It's difficult to hold up the roleplay factor with JUST flavortext in the character generation menu. The rest would be on the player's imagination. This IS a hardcore game and there are plenty of people who roleplay. There are even quite a few that do it successfully, even with a low volume playerbase game like this one.

Let's consider FoodPerson now. What does it have in common with the shelter? They're both buildings, they're both modifications of the submap coordinates to facilitate their environments. They both have flavortext in the main menu. What else does FoodPerson have? It follows up with it on the physical and visual aspects to immerse the player in the environment. You're in the Food Place Break Room. You have items in your inventory(If you choose the designated True Food Person profession) and in the restaurant that compound upon the flavortext to enhance the experience. More specifically, you have a mask that lets you exit to begin your escape vector. You have items that provide a VERY high success chance of escape and safety.

What happens after you escape? That's it, the end. So to address the nihilist approach: Why bother including it if it provides nothing substantially unique? It does. It's meant for people who want to roleplay or have variety in submap coordinate(technically anyway, they don't know that is what it is.). Finally to really drive it home. It only allows 2 professions to choose from. Why? What makes it stand out that it should only have 2? Why can't I be a lumperjack? Because a lumberjack wouldn't normally be working at a fast food chain. It's arguable that the survivor (Unemployed profession) would be there for an interview.

Lastly, why bother having a wilderness scenario? It doesn't change anything right? It has ZERO guided roleplay. Literally it is just a submap coordinate modification. This is what we have:

"You find yourself amongst trees.  The screaming and the moaning is fainter this far from civilization, but you'd better know what you're doing out here.",

How did they find themselves among the trees? Why does this scenario exist? Why bother including this? Variety is the reason. Choice. Roleplay. The player does not need to be guided into roleplay with just flavortext alone, or physical items. Environment plays a large factor in roleplay and gameplay choices. That is where your uniqueness comes from. It is a combination or lack of that creates the unique experience.

Environment matters to roleplayers. Environment matters to be able to compound on things.

The environmental hazard is simply a benefit that is inherited by using a flag that gets this scenario closer to canon. It is not how it should be classified.

I don't think I need to dig any further than this. I certainly don't think this qualifies as a REALLY BAD DAY scenario at all. This is incredibly tame compared to that scenario. Personally I'd rather play naked and afraid profession in a lab challenge than REALLY BAD DAY. There's a reason there's not much coverage on media platforms like YouTube of that scenario. It is a nightmare for presentation and a nightmare to play and it is not fun. It has no longevity other than to try to kill the player with minimal or no chance of survival. It is just compounding artificial difficulty on top of itself and passing it off as challenge. There is zero immersion to it. It is the equivalent of just increasing a boss' health 100x to drag out the fight and pass it off as content. Or to increase the boss' damage by 100x and call that content but not giving the player genuine, fun tools to win.

That is not the goal of this scenario. I can see how you drew that conclusion as I made it the first part of my argument. Those were just the 2 things that I was very excited about as it was natively supported and I didn't have to work to hard to fulfill 2 purposes.

**Comment by LucasHoage** [comment]
After considering public and private feedback, this PR is closed until I've had time to address specific issues. Will return at a later date under a different PR.
